dev/NumberBox/NumberBoxParser.cpp

```diff
@@ -0,0 +1,295 @@
+            }
+            opstack.push(token.str[0]);
+        }
+        // Left paren goes to operator stack
+        else if (token.type == MathToken::TokenType::ParenLeft)
+        {
+            opstack.push('(');
+        }
+        // Right Paren - Pop operators 
+        else if (token.type == MathToken::TokenType::ParenRight)
+        {
+            while ( !opstack.empty() && opstack.top() != '(')
+            {
+                // Pop non leftParen operators onto output
+                out << opstack.top() << ' ';
+                opstack.pop();
+            }
+            // Broken Parentheses. Don't Resolve
+            if (opstack.empty())
+                return L"Malformed";
+            // Pop LeftParen and discard
+            opstack.pop();
+        }
+    }
+
+    // Pop everything to output queue
+    while (!opstack.empty())
+    {
+        if (opstack.top() == '(' || opstack.top() == ')')
+        {
+            return L"Malformed";
+        }
+        out << opstack.top() << ' ';
+        opstack.pop();
+    }
+
+    return out.str();
+}
+
+// Evaluates a postfix expression
+std::optional<double> NumberBoxParser::ComputeRpn(const std::wstring& expr)
+{
+    if (expr == L"Malformed")
+    {
+        return {};
+    }
+    std::stack<double> stack;
+    MathTokenizer input(expr);
+    MathToken token;
+    while ((token = input.GetToken()).type != MathToken::TokenType::EOFToken)
+    {
+        if (token.type == MathToken::TokenType::Operator)
+        {
+            if (stack.empty())
+                return {};
+            double op1 = stack.top();
+            stack.pop();
+
+            if (stack.empty())
+                return {};
+            double op2 = stack.top();
+            stack.pop();
+
+            double res = 0;
+
+            switch (token.str[0])
+            {
+                case '-':
+                    res = op2 - op1;
+                    break;
+                case '+':
+                    res = op1 + op2;
+                    break;
+                case '*':
+                    res = op1 * op2;
+                    break;
+                case '/':
+                    if (op1 == 0)
+                    {
+                        return NAN;
+                    }
+                    res = op2 / op1;
+                    break;
+                case '^':
+                    res = std::pow(op2, op1);
+                    break;
+                default: //uh-oh
+                    return {};
+            }
+            stack.push(res);
+        }
+        else if (token.type == MathToken::TokenType::Numeric) {
+            // Convert to double and push to stack
+            std::wstringstream wss(token.str);
+            double parsed;
+            wss >> parsed;
+            stack.push(parsed);
+        }
+    }
+    return stack.top();
+}
+
+std::optional<double> NumberBoxParser::Compute(const winrt::hstring& expr)
+{
+    std::wstring_view InputAsString = std::wstring_view(expr);
```

Comment:
This function signature can just take wstring_view -- the conversion is implicit and doesn't require an explicit cast. (And if we did need a cast then [static_cast would be preferred](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast))